### ISSUE TITLE
S3626: Jump statements should not be redundant -> Competes with Resharper SeparateLocalFunctionsWithJumpStatement

### ISSUE THREAD

**Issue by naefp** [SonarSource/sonar-dotnet#8536]
### Description

This is not really a false positive, but just an idea to align with Resharper inspections.

[S3626](https://sonarsource.github.io/rspec/#/rspec/S3626/csharp) competes with the Resharper inspection [SeparateLocalFunctionsWithJumpStatement](https://www.jetbrains.com/help/rider/SeparateLocalFunctionsWithJumpStatement.html) "Add explicit return before local function".

Sonar might consider accepting redundant jump statements before local functions as well. 

### Repro steps

Create a local function within any method or loop. Resharper will advise to add an explicit return and place all local functions after that. Sonar will then advise to remove the redundant jump statement.

```c#
private void Some()
{
    DoSomething();
    return;

    void DoSomething()
    {
        // Something
    }
}
```

### Expected behavior

Not competing with Resharper.

### Actual behavior

Resharper and Sonar telling me exactly the opposite to do on that specific code snippet.

### Known workarounds

Disable either Sonar or Resharper rule.

### Related information

* C# 12
* Rider 2023.3.2
* .Net 8
* SonarLint plugin 10.2.1.77304 connected to SonarCloud
* Windows 11

**Comment by pavel-mikula-sonarsource** [comment]
Hi, 

As I see it, our rule is more broad than the local-function specific from ReSharper that solves one corner case.

The superfluous `return;` might cause more confusion IMO, as it goes against the natural syntax of the language. For me, it raises a question about why it's there, when it doesn't do anything. 

So I don't foresee us creating an exception here, and then explaining to all users why the exception exists in this case.

**Comment by antoinebj** [comment]
Apparently there was a change of mind a couple of months ago:
https://community.sonarsource.com/t/c-s3626-false-positive-when-returning-before-local-functions/105102/7

How can we track the progress of this, when can we expect the "fix"?

**Comment by pavel-mikula-sonarsource** [comment]
The issue is now tracked in https://sonarsource.atlassian.net/browse/NET-1149, which is not a public project, unfortunately :/
